pkg/transportrequest/uploadcts.go

```diff
@@ -0,0 +1,233 @@
+	}
+
+	deployStatement, err := getFioriDeployStatement(action.TransportRequestID, action.ConfigFile, action.Application, action.Connection)
+	if err != nil {
+		return err
+	}
+
+	cmd = append(cmd, deployStatement)
+
+	return command.RunShell("/bin/bash", strings.Join(cmd, "\n"))
+}
+
+func getPrepareFioriEnvironmentStatement(deps []string, npmInstallOpts []string) string {
+	cmd := []string{
+		"npm",
+		"install",
+		"--global",
+	}
+	cmd = append(cmd, npmInstallOpts...)
+	cmd = append(cmd, deps...)
+	return strings.Join(cmd, " ")
+}
+
+func getFioriDeployStatement(
+	transportRequestID string,
+	configFile string,
+	app CTSApplication,
+	cts CTSConnection,
+) (string, error) {
+	desc := app.Desc
+	if len(desc) == 0 {
+		desc = "Deployed with Piper based on SAP Fiori tools"
+	}
+
+	useConfigFile, noConfig, err := handleConfigFile(configFile)
+	if err != nil {
+		return "", err
+	}
+	cmd := []string{
+		"fiori",
+		"deploy",
+		"--failfast", // provide return code != 0 in case of any failure
+		"--yes",      // autoconfirm --> no need to press 'y' key in order to confirm the params and trigger the deployment
+		"--username", abapUserKey,
+		"--password", abapPasswordKey,
+		"--description", fmt.Sprintf("\"%s\"", desc),
+	}
+
+	if noConfig {
+		cmd = append(cmd, "--noConfig") // no config file, but we will provide our parameters
+	}
+	if useConfigFile {
+		cmd = append(cmd, "--config", fmt.Sprintf("\"%s\"", configFile))
+	}
+	if len(cts.Endpoint) > 0 {
+		log.Entry().Debugf("Endpoint '%s' used from piper config", cts.Endpoint)
+		cmd = append(cmd, "--url", cts.Endpoint)
+	} else {
+		log.Entry().Debug("No endpoint found in piper config.")
+	}
+	if len(cts.Client) > 0 {
+		log.Entry().Debugf("Client '%s' used from piper config", cts.Client)
+		cmd = append(cmd, "--client", cts.Client)
+	} else {
+		log.Entry().Debug("No client found in piper config.")
+	}
+	if len(transportRequestID) > 0 {
+		log.Entry().Debugf("TransportRequestID '%s' used from piper config", transportRequestID)
+		cmd = append(cmd, "--transport", transportRequestID)
+	} else {
+		log.Entry().Debug("No transportRequestID found in piper config.")
+	}
+	if len(app.Pack) > 0 {
+		log.Entry().Debugf("application package '%s' used from piper config", app.Pack)
+		cmd = append(cmd, "--package", app.Pack)
+	} else {
+		log.Entry().Debug("No application package found in piper config.")
+	}
+	if len(app.Name) > 0 {
+		log.Entry().Debugf("application name '%s' used from piper config", app.Name)
+		cmd = append(cmd, "--name", app.Name)
+	} else {
+		log.Entry().Debug("No application name found in piper config.")
+	}
+
+	return strings.Join(cmd, " "), nil
+}
+
+func getSwitchUserStatement(user string) string {
+	return fmt.Sprintf("su %s", user)
+}
+
+func handleConfigFile(path string) (useConfigFile, noConfig bool, err error) {
```

Comment:
Can you think of a more telling function name than `handleConfigFile`? 

I'm not so happy with this function in general. My gut tells me this must be possible to do in a simpler way. How about splitting the functionality into to functions: One for `noConfig` and one for `useConfigFile`? I don't see the two being dependent on each other.

Reply:
>Can you think of a more telling function name than `handleConfigFile`?

Do you have any proposal?

> How about splitting the functionality into to functions

And I thought it would be better to understand to have it in one function ...

> I don't see the two being dependent on each other.

Both parameters rely on the config file

Reply:
Would this not do the same for the useConfigFile functionality?

```
func useConfigFile(path string) (exists bool, err error){
	if len(path) != 0 {
		exists, err = files.FileExists(path)
	}
	return
}
```

Reply:
I'm currently looking into the noConfig functionality...

Reply:
This should do the same for the noConfig: 
```
func noConfig(path string) (noConfig bool, err error) {
	exists := false

	if len(path) == 0 {
		exists, err = files.FileExists(defaultConfigFileName)
	} else {
		exists, err = files.FileExists(path)
		if exists == false && path != defaultConfigFileName {
			err = fmt.Errorf("configured deploy config file '%s' does not exists", path)
			return
		}
	}
	noConfig = !exists

	return
}
```